src/Valit/ValitRules.cs

```diff
@@ -119,23 +129,41 @@ IValitResult IValitRules<TObject>.Validate(Func<IValitRule<TObject>, bool> predi
         private IValitResult Validate(IEnumerable<IValitRule<TObject>> rules)
         {
             var result = ValitResult.Success;
-
             foreach(var rule in rules.ToList())
```

Comment:
Not something that is changed in this PR, but why is the `.ToList()` needed here?

Reply:
to avoid multiple enumerations